### PR TITLE
feat(cx-2438): sort and filter for high demand index

### PR DESCRIPTION
**Pull request by lordkiz**
This PR resolves [CX-2438]

### Description

https://user-images.githubusercontent.com/18648835/165268042-fbe4d8a6-b3b4-40f4-86ee-2f58782972cd.mp4

### PR Checklist

- [x] I tested my changes on **iOS** / **Android**.
- [x] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- sort and filter for high demand index - kizito

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look on our [docs], or get in touch with us.

[app state migration]: /docs/adding_state_migrations.md
[feature flag]: /docs/developing_a_feature.md
[docs]: /docs/README.md

[CX-2438]: https://artsyproduct.atlassian.net/browse/CX-2438?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/app/Scenes/MyCollection/utils/localArtworkSortAndFilter.ts [hunk 1]
```diff
@@ -93,6 +93,24 @@ export const useLocalArtworkFilter = (artworksList?: any[] | null) => {
         // tslint:disable-next-line: no-shadowed-variable
         localSortAndFilter: (artworks) => orderBy(artworks, (a) => a.artistNames, "desc"),
       },
+      {
+        paramName: FilterParamName.sort,
+        displayText: "Demand Index (High to low)",
```
Comment (olerichter00) [inline src/app/Scenes/MyCollection/utils/localArtworkSortAndFilter.ts:98]:
```suggestion
        displayText: "Demand Index (High to Low)",
```

**Review by olerichter00 — APPROVED** [review]
Looks good! Feel free to merge after fixing the copy 🌟

File: src/app/Scenes/MyCollection/utils/localArtworkSortAndFilter.ts [hunk 2]
```diff
@@ -93,6 +93,24 @@ export const useLocalArtworkFilter = (artworksList?: any[] | null) => {
...
+      },
+      {
+        paramName: FilterParamName.sort,
+        displayText: "Demand Index (Low to High)",
+        paramValue: "demand-index-low-to-high",
+        // tslint:disable-next-line: no-shadowed-variable
+        localSortAndFilter: (artworks) => {
+          console.log("artworks", artworks)
+          return orderBy(artworks, (a) => a.marketPriceInsights?.demandRank ?? 0, "asc")
+        },
```
Comment (MrSltun) [inline src/app/Scenes/MyCollection/utils/localArtworkSortAndFilter.ts:112]:
```suggestion
        localSortAndFilter: (artworks) =>
          orderBy(artworks, (a) => a.marketPriceInsights?.demandRank ?? 0, "asc"),
```

**Review by MrSltun — APPROVED** [review]
Great job Kizito 👏 👏 
I just left a small suggestion :)

**Review by MrSltun — APPROVED** [review]

**Pull request merged**